apps/src/templates/HideToolbarHelper.jsx

```diff
@@ -0,0 +1,144 @@
+    this.wasToolbarShowing = false;
+
+    // ...but only do these things once.
+    this.didTrackToolbarHide = false;
+
+    if (window.location.search.indexOf('force_show_toolbar_helper') !== -1) {
+      cookies.remove(HideToolbarHelperCookieName, {path: '/'});
+    }
+  }
+
+  isiOS13() {
+    return navigator.userAgent.indexOf('iPhone OS 13') !== -1;
+  }
+
+  isHideCookieSet() {
+    return cookies.get(HideToolbarHelperCookieName);
+  }
+
+  isLandscape() {
+    return window.orientation !== 0;
+  }
+
+  isToolbarShowing() {
+    return window.innerHeight < document.body.offsetHeight;
+  }
+
+  updateLayout = () => {
+    const isiOS13 = this.isiOS13();
+    const isHideCookieSet = this.isHideCookieSet();
+    const isLandscape = this.isLandscape();
+
+    // window.innerHeight is smaller than document.body.offsetHeight when
+    // the iOS 13 Safari toolbar is showing.  It becomes equal when the toolbar
+    // is hidden.  (Interestingly, document.documentElement.clientHeight is
+    // the same as document.body.offsetHeight in both cases.)
+    const isToolbarShowing = this.isToolbarShowing();
+
+    if (
+      this.wasToolbarShowing &&
+      !isToolbarShowing &&
+      !this.didTrackToolbarHide
+    ) {
+      // Let's track the disapearance of the toolbar.
+      trackEvent('Research', 'HideToolbarHelper', 'hid-' + window.innerHeight);
+
+      // And also set the cookie so this user doesn't see the helper again
+      // for a year.
+      this.setHideHelperCookie();
+
+      this.didTrackToolbarHide = true;
+    }
+
+    this.wasToolbarShowing = isToolbarShowing;
+
+    const showHelper =
+      isiOS13 && !isHideCookieSet && isLandscape && isToolbarShowing;
+
+    this.setState({showHelper});
+  };
+
+  componentDidMount() {
+    this.updateLayout();
+
+    this.updateLayoutListener = _.throttle(this.updateLayout, 200);
+    window.addEventListener('orientationchange', this.updateLayoutListener);
+
+    /* These two events catch all viewport changes. */
+    window.addEventListener('resize', this.updateLayoutListener);
+    window.addEventListener('scroll', this.updateLayoutListener);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener(
+      'orientationchange',
+      this.updateLayoutListener,
+      false
+    );
+    document.removeEventListener('resize', this.updateLayoutListener, false);
+    document.removeEventListener('scroll', this.updateLayoutListener, false);
+  }
+
+  onClick = () => {
+    this.setHideHelperCookie();
+    this.updateLayout();
+
+    // Let's track the click-to-dismiss event.
+    trackEvent(
+      'Research',
+      'HideToolbarHelper',
+      'click-' + window.innerHeight + '-' + document.body.offsetHeight
+    );
+  };
+
+  setHideHelperCookie() {
+    cookies.set(HideToolbarHelperCookieName, 'true', {expires: 365, path: '/'});
+  }
+
+  render() {
+    if (this.state.showHelper) {
+      return (
+        <div className="hide_toolbar_helper" onClick={this.onClick}>
```

Comment:
Per the conversation at https://codedotorg.slack.com/archives/C0T0PNR0D/p1588014772038400, please migrate this to use the https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/Tooltip.jsx component or another existing tooltip component.

Reply:
After our long discussion, it turns out this is more of a "callout" than a "tooltip" from our point of view, so I will merge as-is and we can work on factoring it out, for shared use, next.